commons/zenoh-config/src/lib.rs

```diff
@@ -97,6 +97,19 @@ pub struct DownsamplingItemConf {
     pub flow: DownsamplingFlow,
 }
 
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub struct ConnectionRetryConf {
+    //TODO(sashacmc): add comments, and infinit value for counter
+    // ModeDependentValue:
+    // 1) only for exit_on_fail
+    // 2) for all
+    // 3) for no one (we have connect and listen config, they already separate)
+    pub count: u32,
+    pub timeout_ms: u64,
+    pub timeout_increase_factor: f32,
+    pub exit_on_fail: ModeDependentValue<bool>,
```

Comment:
Not sure why we need ModeDependentValue, it increase complexity, but I don't see profit if we have two separate retry config for listen and connect.